packages/slate/src/create-editor.ts

```diff
@@ -25,11 +25,23 @@ export const createEditor = (): Editor => {
   const editor: Editor = {
     children: [],
     operations: [],
+    nativeOperationsQueue: [],
     selection: null,
     isInline: () => false,
     isVoid: () => false,
     onChange: () => {},
+    flushQueuedNativeOperations: () => {
```

Comment:
I'd like to achieve this without exposing any user-facing pieces, so that it's completely under the covers, and could even be eliminated one day if browser behavior improves.

I believe this would be possible by using the `WeakMap` pattern that's used elsewhere in the codebase to keep track of internal state. In this case we'd probably need two:

```ts
const AS_NATIVE: WeakMap<Editor, boolean> = new WeakMap()
const NATIVE_OPERATIONS: WeakMap<Editor, Operation[]> = new WeakMap()
```

With an associated helper:

```js
asNative(() => {
  editor.exec(...)
})
```

Such that it ensures all operations that happen inside that function are queued as native operations. And the queue can be checked still in `apply`.

They should prevent having to add any new functions to the `Editor` interface, or exposing any `native?:` flags to transforms/operations as well.